src/python/pants/init/engine_initializer.py

```diff
@@ -369,6 +370,10 @@ def setup_legacy_graph_extended(
 
         symbol_table = _legacy_symbol_table(build_file_aliases)
 
+        # TODO: register this with the SymbolTable/LegacyPythonCallbacksParser so that the aliases
```

Comment:
It feels like doing this in this PR might make sense? Or is it usable in some other way right now?

Reply:
I don't think so. This will be non-trivial to do, e.g. John pointing out that we need to handle `BuildFileAliases` and `targets2` registering the same symbols. Who should win? Should we error? (I vote error). 

The purpose of this PR is to a) expose the `targets2` endpoint and b) allow going from `HydratedStruct -> WrappedTarget`, so that we can in the followup request `Targets` and start using the Target API in goal rules.